src/spikeinterface/core/core_tools.py

```diff
@@ -934,3 +941,72 @@ def convert_bytes_to_str(byte_value: int) -> str:
         byte_value /= 1024
         i += 1
     return f"{byte_value:.2f} {suffixes[i]}"
+
+
+def spike_vector_to_dict(spike_vector: np.ndarray) -> list[dict]:
+    """
+    Computes all spike trains for all units/segments from a spike vector.
+
+    Parameters
+    ----------
+    spike_vector: np.ndarray
+        The spike vector to convert.
+
+    Returns
+    -------
+    spike_trains: list[dict]:
+        A list containing, for each segment, the spike trains of all units.
+    """
+    assert HAVE_NUMBA, "spike_vector_to_dict() requires `numba`!"
+
+    spike_trains = _vector_to_dict(
+        spike_vector["sample_index"].astype(np.int64),
+        spike_vector["unit_index"].astype(np.int64),
+        spike_vector["segment_index"].astype(np.int64),
+    )
+
+    return spike_trains
+
+
+if HAVE_NUMBA:
+
+    @numba.jit((numba.int64[::1], numba.int64[::1], numba.int64[::1]), nopython=True, nogil=True, cache=True)
+    def _vector_to_dict(sample_index, unit_index, segment_index):
+        """
+        Fast method to convert a spike vector into spike train for all segments and units.
+
+        Parameters
+        ----------
+        sample_index: array
+            spike_vector["sample_index"]
+        unit_index: array
+            spike_vector["unit_index"]
+        segment_index: array
+            spike_vector["segment_index"]
+
+        Returns
+        -------
+        spike_trains: list[dict]
+            A list containing, for each segment, the spike trains of all units.
+        """
+
+        spike_trains = []
```

Comment:
It is very confusing that you have a variable named `spike_trains` and then another named `spk_trains` with different type and then one goes inside the other.

The outer one could be called `spike_trains_per_segment`.

Reply:
Done :+1: